### PR TITLE
[PLAT-813] Return number of results in metadata search

### PR DESCRIPTION
**Pull request by amvertex**
## Summary
We want to show the user the number of results from a metadata search in the scene tree.

## Test Plan
Will be tested with connect-app PR

## Release Notes
None

## Possible Regressions
None

## Dependencies
None

### CODE REVIEW

File: packages/viewer/src/components/scene-tree/lib/controller.ts [hunk 1]
```diff
@@ -756,7 +780,23 @@ export class SceneTreeController {
         );
         const rows = [...start, ...fetchedRows, ...end, ...fill];
 
-        this.updateState({ ...this.state, totalRows, rows });
+        const rowsWithFilterHit = rows.filter((row) => row?.node?.filterHit);
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/components/scene-tree/lib/controller.ts:783]: Will this work with pagination?

File: packages/viewer/src/components/scene-tree/lib/controller.ts [hunk 2]
```diff
@@ -121,8 +122,15 @@ export class SceneTreeController {
    */
   public onStateChange = new EventDispatcher<SceneTreeState>();
 
+  /**
+   * A dispatcher that emits an event whenever the number of search results has changed.
+   */
+  public numberOfRowsWithFilterHitChanged = new EventDispatcher<number>();
```
Comment (danschultz) [inline packages/viewer/src/components/scene-tree/lib/controller.ts:128]: Looks like this value is contained within state. Can we just rely on the `onStateChange` dispatcher?

File: packages/viewer/src/components/scene-tree/lib/controller.ts [hunk 3]
```diff
@@ -496,14 +515,24 @@ export class SceneTreeController {
...
+          const req = new FilterRequest();
+          req.setFilter(term);
+          req.setFullTree((options.includeCollapsed ?? true) === true);
+          if (options.columns) req.setColumnsKeysList(options.columns);
+
+          this.client.filter(req, metadata, handler);
+        }
+      );
+
+      const { numberOfResults } = res.toObject();
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/components/scene-tree/lib/controller.ts:531]: Can we add a test for this? Otherwise, looks good to me.

Comment (amvertex) [reply packages/viewer/src/components/scene-tree/lib/controller.ts:531]: Good idea 👍

File: packages/viewer/src/components/scene-tree/lib/controller.ts [hunk 4]
```diff
@@ -41,6 +42,7 @@ export type JwtProvider = () => string | undefined;
 
 export interface SceneTreeState {
   totalRows: number;
+  numberOfRowsWithFilterHit: number;
```
Comment (MadisonEhlers-Vertex) [inline packages/viewer/src/components/scene-tree/lib/controller.ts:45]: Can this match the naming convention of `totalRows`? Maybe `totalFilteredRows`?

**Review by MadisonEhlers-Vertex — APPROVED** [review]

**Pull request merged**